provider crashed when cloning LXC container

When cloning an existing CT container the provider crashes with error:

Stack trace from the terraform-provider-proxmox_v0.46.3 plugin:
panic: interface conversion: interface {} is types.CustomBool, not bool
Error: The terraform-provider-proxmox_v0.46.3 plugin crashed!

**To Reproduce**
run following terraform script:


```hcl

resource "proxmox_virtual_environment_container" "k3s_master" {
    count = 1
    node_name = "proxmox01"
    tags      = ["kubernetes"]
    description = "k3s-master-nodes${count.index + 1}"
    vm_id     = count.index + 5000

    clone {
      datastore_id = "local"
      vm_id = 9001
    }

}

```

**Expected behavior**
The ct container should be clones without issues

- Single or clustered Proxmox: cluster but against single host
- Provider version (ideally it should be the latest version): v0.46.3
- Terraform version: v1.7.3
- OS (where you run Terraform from): Linux Ubuntu

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -8,6 +8,8 @@
         "iothread",
         "keyctl",
         "mbps",
+        "nameserver",
+        "nestif",
         "nolint",
         "NUMA",
         "proxmoxtf",

--- a/proxmoxtf/resource/container.go
+++ b/proxmoxtf/resource/container.go
@@ -1512,12 +1512,14 @@ func containerCreateCustom(ctx context.Context, d *schema.ResourceData, m interf
 		rateLimit := networkInterfaceMap[mkResourceVirtualEnvironmentContainerNetworkInterfaceRateLimit].(float64)
 		vlanID := networkInterfaceMap[mkResourceVirtualEnvironmentContainerNetworkInterfaceVLANID].(int)
 		mtu := networkInterfaceMap[mkResourceVirtualEnvironmentContainerNetworkInterfaceMTU].(int)
+		firewall := networkInterfaceMap[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall].(bool)
 
 		if bridge != "" {
 			networkInterfaceObject.Bridge = &bridge
 		}
 
 		networkInterfaceObject.Enabled = enabled
+		networkInterfaceObject.Name = name
 
 		if len(initializationIPConfigIPv4Address) > ni {
 			if initializationIPConfigIPv4Address[ni] != "" {
@@ -1537,11 +1539,13 @@ func containerCreateCustom(ctx context.Context, d *schema.ResourceData, m interf
 			}
 		}
 
+		if firewall {
+			networkInterfaceObject.Firewall = types.CustomBool(firewall).Pointer()
+		}
+
 		if macAddress != "" {
 			networkInterfaceObject.MACAddress = &macAddress
 		}
-
-		networkInterfaceObject.Name = name
 
 		if rateLimit != 0 {
 			networkInterfaceObject.RateLimit = &rateLimit
@@ -1739,15 +1743,17 @@ func containerGetExistingNetworkInterface(
 
 		networkInterface := map[string]interface{}{}
 
+		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceEnabled] = true
+		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceName] = nv.Name
+
 		if nv.Bridge != nil {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceBridge] = *nv.Bridge
 		} else {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceBridge] = ""
 		}
 
-		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceEnabled] = true
-		if nv.Firewall != nil {
-			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall] = *nv.Firewall
+		if nv.Firewall != nil && *nv.Firewall {
+			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall] = true
 		} else {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall] = false
 		}
@@ -1757,8 +1763,6 @@ func containerGetExistingNetworkInterface(
 		} else {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceMACAddress] = ""
 		}
-
-		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceName] = nv.Name
 
 		if nv.RateLimit != nil {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceRateLimit] = *nv.RateLimit
@@ -2287,16 +2291,17 @@ func containerRead(ctx context.Context, d *schema.ResourceData, m interface{}) d
 
 		networkInterface := map[string]interface{}{}
 
+		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceEnabled] = true
+		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceName] = nv.Name
+
 		if nv.Bridge != nil {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceBridge] = *nv.Bridge
 		} else {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceBridge] = ""
 		}
 
-		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceEnabled] = true
-
-		if nv.Firewall != nil {
-			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall] = *nv.Firewall
+		if nv.Firewall != nil && *nv.Firewall {
+			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall] = true
 		} else {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceFirewall] = false
 		}
@@ -2306,8 +2311,6 @@ func containerRead(ctx context.Context, d *schema.ResourceData, m interface{}) d
 		} else {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceMACAddress] = ""
 		}
-
-		networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceName] = nv.Name
 
 		if nv.RateLimit != nil {
 			networkInterface[mkResourceVirtualEnvironmentContainerNetworkInterfaceRateLimit] = *nv.RateLimit

